### DISCUSSION TITLE
Why I shouldn't baseline my dev environment?

### DISCUSSION THREAD

**Discussion by igolka97** [prisma/prisma#8322] [Q&A] [answered]
It looks like I doing something wrong.

What I did:
1. Dumped my PSQL DB.
2. `prisma migrate dev` then my db data was lost
3. Restore db from dump with `clean before restore` flag
4. `prisma migrate resolve --applied 20210716173251_initial_migration`

As a result i have working Prisma Migrate and no any data loss. Looks like everything works well, is't it? What disadvantages I can got working in baselined dev env?

May be I'm looking for deeper understanding how `prisma migrate dev` works

**Comment by janpio** [comment, marked as answer]
Baselining makes sure Migrate knows about the current existing state of the database, and so sees no need to reset the database (which makes you lose your data).

**Reply by igolka97** [reply]
Still not clear to me. So why [docs](https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/add-prisma-migrate-to-a-project#apply-the-initial-migrations) force me to reset dev environment?

What proper way to set baseline state of db without loosing data? 

May be I should replace initial migration with db schema dump?

**Reply by janpio** [reply]
Because the docs tell you that if you have data you can not lose, you should baseline the database. Only on development databases wher you don't you should use a reset. 
So why a reset? A reset is much more predictable and will make sure everything is 100% in exactly the way Prisma would have created it. Baselining does not.

**Reply by igolka97** [reply]
So my main question and fear — what disadvantages I could face if I use baselined db at dev and what if my `initial_migration` is just schema dump from `pg_dump`? It's okay and nothing to worry about?

AFAIK [baselining](https://odetocode.com/blogs/scott/archive/2008/01/31/versioning-databases-the-baseline.aspx) — setting zero point of DB schema. Does it matter how I'll get it?

**Reply by tomhoule** [reply]
Baselining will work in development, but it is discouraged because at some point, you will have to reset the data to get to a clean state (e.g. when you edit a migration, switch between branches with different migrations, etc.). The data can be part of a `seed.ts` script to always have automatically it populated.

**Reply by igolka97** [reply]
Thanks so much! Now I going to have a challenge to seed my DB from dump with bash script. The difficulty is that I should seed World Geo data via Docker. It's too much to seed with plain code.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
